utils/error_injection.hh

```diff
@@ -454,6 +462,18 @@ public:
         co_await func(handler);
     }
 
+    // \brief Inject "breakpoint"
+    // Injects a pause in the code execution that's woken up explicitly by the injector
+    // request
+    // \param wfm -- the wait_for_message instance that describes details of the pause
```

Comment:
Thanks for the comment, and I like the word "breakpoint" (I wonder why we need to call all these overloads "inject" and can't call them with other names, like breakpoint). But I would have liked to see more information and even an example. What does the "wait_for_message" suppose to contain (example code?). How does the user actually send this "message" in test code?

Reply:
Like this

```python
    await api.enable_injection(server.ip_addr, injection_name) # activate injection
    log = manager.server_open_log(server.server_id)
    await log.wait_for(corresponding_log_message) # wait for scylla to step on it and start waiting
    do_something_while_scylla_is_paused()
    await api.message_injection(server.ip_addr, injection_name) # notify scylla that it can continue working
```

you can check e.g. `test/topology_custom/test_major_compaction.py::test_shutdown_drain_during_compaction` or `test/topology_custom/test_raft_no_quorum.py::test_quorum_lost_during_node_join` for a working code


Regarding your question

> What does the "wait_for_message" suppose to contain (example code)?

The answer is -- nothing, these "messages" come without payload by initial design

Reply:
I merged the current version, but please consider including some of this wisdom later in a comment, it's not easy to guess.

To be honest, I'm still not even sure I understand how this works... What if there are concurrently 100 fibers in 10 shards all waiting on the same breakpoint injection, will a single message wake up all of them? One of them?

Why did we even call this a "message" (since, as you said, this message can't have any content) instead of "signal" or "wakeup"?

Reply:
Oh, and I really don't like the log.wait_for() in the above example. There has to be a better way... We should have had a REST API, like the one that sends a message, to wait until the breakpoint was reached.

Of course, we can do all of this later. But someone will need to care :-( Last time I wanted to use injections I found myself faced with a dozen options of what to use, each one looks more half-baked than the others :-(